app/javascript/components/student/ExerciseStatusChart.tsx

```diff
@@ -52,6 +52,18 @@ export const ExerciseStatusChart = ({
 
 const DEFAULT_ERROR = new Error('Unable to load information')
 
+const LoadingComponent = () => {
+  const [isShowing, setIsShowing] = useState(false)
+
+  setTimeout(() => {
+    setIsShowing(true)
+  }, 200)
```

Comment:
On [this page](https://upmostly.com/tutorials/settimeout-in-react-components-using-hooks), they suggest to use a `useEffect` hook for setting and clearing timeouts.

Reply:
Ah yes! Good catch.